itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationOnHDFSEncryptedZones.java

```diff
@@ -119,16 +120,15 @@ public void targetAndSourceHaveDifferentEncryptionZoneKeys() throws Throwable {
             .run("insert into table encrypted_table values (2,'value2')")
             .dump(primaryDbName, dumpWithClause);
 
-    replica
-        .run("repl load " + primaryDbName + " into " + replicatedDbName
-                + " with('hive.repl.add.raw.reserved.namespace'='true', "
-                + "'hive.repl.replica.external.table.base.dir'='" + replica.externalTableWarehouseRoot + "', "
-                + "'distcp.options.pugpbx'='', 'distcp.options.skipcrccheck'='')")
-        .run("use " + replicatedDbName)
-        .run("repl status " + replicatedDbName)
-        .verifyResult(tuple.lastReplicationId)
-        .run("select value from encrypted_table")
-        .verifyFailure(new String[] { "value1", "value2" });
+    try {
+      replica.run("repl load " + primaryDbName + " into " + replicatedDbName
+          + " with('hive.repl.add.raw.reserved.namespace'='true', " + "'hive.repl.replica.external.table.base.dir'='"
+          + replica.externalTableWarehouseRoot + "', "
+          + "'distcp.options.pugpbx'='', 'distcp.options.skipcrccheck'='')");
+      Assert.fail("Test should have thrown an exception because cross-encryption-zone is not allowed for RAW");
+    } catch (IOException ioe) {
+      // ignore
```

Comment:
Hey @belugabehr  just read the detailed comment on the JIRA about this but I believe we should add some explanation here as well for clarity